File: tests/output/go/tests/manual/forge_secured_user_token_test.go
package manual

import (
	"crypto/hmac"
	"crypto/sha256"
	"encoding/base64"
	"encoding/json"
	"strings"
	"testing"
	"time"

	"github.com/stretchr/testify/require"

	agentStudio "github.com/algolia/algoliasearch-client-go/v4/algolia/agent-studio"
)

func TestForgeSecuredUserToken(t *testing.T) {
	client, err := agentStudio.NewClient("appID", "apiKey")
	require.NoError(t, err)

	token, err := client.ForgeSecuredUserToken("my-secret-key", "my-key-id", "user-123")
	require.NoError(t, err)

	parts := strings.Split(token, ".")
	require.Len(t, parts, 3)

	headerJSON, err := base64.RawURLEncoding.DecodeString(parts[0])
	require.NoError(t, err)
	var header map[string]any
	require.NoError(t, json.Unmarshal(headerJSON, &header))
	require.Equal(t, "HS256", header["alg"])
	require.Equal(t, "JWT", header["typ"])
	require.Equal(t, "my-key-id", header["kid"])

	payloadJSON, err := base64.RawURLEncoding.DecodeString(parts[1])
	require.NoError(t, err)
	var payload map[string]any
	require.NoError(t, json.Unmarshal(payloadJSON, &payload))
	require.Equal(t, "user-123", payload["sub"])
	exp := int64(payload["exp"].(float64))
	expectedExp := time.Now().Unix() + 24*3600
	require.InDelta(t, expectedExp, exp, 5)

	mac := hmac.New(sha256.New, []byte("my-secret-key"))
	mac.Write([]byte(parts[0] + "." + parts[1]))
	expectedSig := base64.RawURLEncoding.EncodeToString(mac.Sum(nil))
	require.Equal(t, expectedSig, parts[2])
}

func TestForgeSecuredUserTokenCustomExpiry(t *testing.T) {
	client, err := agentStudio.NewClient("appID", "apiKey")
	require.NoError(t, err)

	token, err := client.ForgeSecuredUserToken("my-secret-key", "my-key-id", "user-456", agentStudio.WithExpiresIn(3600))
	require.NoError(t, err)

	parts := strings.Split(token, ".")
	require.Len(t, parts, 3)

	payloadJSON, err := base64.RawURLEncoding.DecodeString(parts[1])
	require.NoError(t, err)
	var payload map[string]any
	require.NoError(t, json.Unmarshal(payloadJSON, &payload))
	exp := int64(payload["exp"].(float64))
	expectedExp := time.Now().Unix() + 3600
	require.InDelta(t, expectedExp, exp, 5)
}

File: tests/output/php/src/manual/ForgeSecuredUserTokenTest.php
<?php

namespace Algolia\AlgoliaSearch\Tests;

use Algolia\AlgoliaSearch\Api\AgentStudioClient;
use PHPUnit\Framework\TestCase;

/**
 * @internal
 *
 * @coversNothing
 */
class ForgeSecuredUserTokenTest extends TestCase
{
    public function testForgeSecuredUserToken(): void
    {
        $client = AgentStudioClient::create('appID', 'apiKey');

        $token = $client->forgeSecuredUserToken('my-secret-key', 'my-key-id', 'user-123');

        $parts = explode('.', $token);
        $this->assertCount(3, $parts);

        $header = json_decode(base64_decode(strtr($parts[0], '-_', '+/')), true);
        $this->assertSame('HS256', $header['alg']);
        $this->assertSame('JWT', $header['typ']);
        $this->assertSame('my-key-id', $header['kid']);

        $payload = json_decode(base64_decode(strtr($parts[1], '-_', '+/')), true);
        $this->assertSame('user-123', $payload['sub']);
        $this->assertEqualsWithDelta(time() + 24 * 3600, $payload['exp'], 5);

        $expectedSig = rtrim(strtr(base64_encode(hash_hmac('sha256', $parts[0].'.'.$parts[1], 'my-secret-key', true)), '+/', '-_'), '=');
        $this->assertSame($expectedSig, $parts[2]);
    }

    public function testForgeSecuredUserTokenCustomExpiry(): void
    {
        $client = AgentStudioClient::create('appID', 'apiKey');

        $token = $client->forgeSecuredUserToken('my-secret-key', 'my-key-id', 'user-456', 3600);

        $parts = explode('.', $token);
        $payload = json_decode(base64_decode(strtr($parts[1], '-_', '+/')), true);
        $this->assertEqualsWithDelta(time() + 3600, $payload['exp'], 5);
    }
}
